src/main/drivers/bus_i2c_stm32f10x.c

```diff
@@ -132,11 +132,14 @@ void I2C3_EV_IRQHandler(void) {
 }
 #endif
 
-static bool i2cHandleHardwareFailure(I2CDevice device)
+static bool i2cHandleHardwareFailure(I2CDevice device, bool unstick)
```

Comment:
I think it is better to pass in the `state` pointer - more flexible and also avoids a horrible `bool` parameter.